README.md

```diff
@@ -6,44 +6,167 @@
+$form->onSubmit(function($form){
+    $u = $form->model->newInstance()->tryLoadBy('email', $form->model['email']);
+    if ($u->loaded()) {
+        return $form->error('email', 'There is already account with this email.');
+    }
+    $form->model->save();
+    return $form->success('Thank you. You may login now.');
+});
+```
+
+Again - the result is beautiful and interractive:
+
+![](docs/images/register-in-15-lines.png)
+
+## What can be done in few hours?
+
+With some knowledge and training you will be able to build entire business-class applications just within few hours. 
+
+-   Warehouse app - source: https://github.com/atk4/warehouse
+-   Warehouse app - demo: https://atk-warehouse.herokuapp.com (free hosting, so wait 5 seconds)
+
+WARNING: this specification is intentionally kept simple for demonstration purposes.
+
+## Reinventing Data Persistence
+
+Conventionally you have to work with ORMs or direct data queries. Agile UI relies on a entirely different class framework for database persistence - [Agile Data](https://github.com/atk4/data). Not only it has all the benefits of ORM, but it also addresses all the faults and shortcomings to deliver powerful features like:
+
+-   Aggregation, scoping, behaviours, audit and custom actions
+-   NoSQL and RestAPI persistence support for Cloud Databases and Services
+-   SQL Expressions, Deep traversal and working with multiple persistences
+
+Through tight integration with Agile Data it makes it possible for your Forms to automatically explore relations and provide values for drop-downs:
+
+![](./docs/images/relation-exploration.png)
+
+## Add-ons and extensions
+
+Agile UI has been developed from ground-up to be extensible in a whole new way. Each add-on delivers wide range of classes you can incorporate into your application without worrying about UI and Data compatibility.
+
+### Components
+
+Components present a UI element which you can place anywhere in your layout. How about a UI for performing all user management operations - reset password, block account or edit user data:
+
+``` php
+$user_admin = $app->layout->add(new \atk4\login\UserAdmin(new User($db)));
+```
+
+### Grid Actions
+
+What about also adding a button for a popup to send a custom message to a user implemented in entirely different add-on?
+
+``` php
+$user_admin->addAction(\atk4\sendgrid\Actions\SendUserMessage());
+```
+
+### Entire Application Parts
+
+Even entire parts of your applications can be provided by add-on. The following code adds Log-in, Registration and Password reminder for your application:
+
+``` php
+$app->add(new \atk4\login\Auth([
+  new User($db),
+  'reminder'=>true, 'register'=>true
+  'outbox' = new \atk4\sendgrid\Outbox(['key'=>$apiKey])
+]));
+```
+
+![Login](./docs/images/login-demo.png)
+
+### Extending Layouts
+
+An add-on can add new menu items. It does not happen automatically but through a very simple line of code:
+
+``` php
+$auth->addChangePassword($app->layout->menu_user);
```

Comment:
why not vice-versa? `$app->layout->menu_user->addItem($auth->getChangePasswordMenuItem())`